crc.c

```diff
@@ -58,3 +59,29 @@ unsigned short crc16(unsigned char *buf, unsigned int len) {
 	}
 	return cksum;
 }
+
+/**
+  * @brief  Computes the 32-bit CRC of a given buffer of data word(32-bit) using
+  * Hardware Acceleration.
+  * @param  pBuffer: pointer to the buffer containing the data to be computed
+  * @param  BufferLength: length of the buffer to be computed
+  * @retval 32-bit CRC
+  */
+uint32_t crc32(uint32_t *pBuffer, uint32_t BufferLength) {
+  uint32_t index = 0;
+
+  for(index = 0; index < BufferLength; index++) {
+    CRC->DR = pBuffer[index];
+  }
+  return (CRC->DR);
+}
+
+/**
+  * @brief  Resets the CRC Data register (DR).
+  * @param  None
+  * @retval None
+  */
+void crc32_reset(void) {
+  /* Reset CRC generator */
+  CRC->CR = CRC_CR_RESET;
```

Comment:
(rest of CRC_CR is reserved, this should be `CRC->CR |= CRC_CR_RESET;`

Reply:
I copied the function from ST library v1.6.0. It doest hurt if I make the change, thanks